### DISCUSSION TITLE
MIME Type conflict running the VueJS + bootstrap-vue example

### DISCUSSION THREAD

**Discussion by ul-gh** [TotallyInformation/node-red-contrib-uibuilder#188] [Q&A] [answered]
Hi,

running node-red Version: 2.1.6-2 from Debian packages (Raspberry Pi),
installed uibuilder 5.1.1, from the Node Red pallette,
installed vue 3.2.38, bootstrap-vue 2.22.0 and bootstrap 4.6.2 from the uibuilder "Libraries" tab:

The VueJS + bootstrap-vue basic example does not run.

On the Firefox developer console I get a MIME-Type conflict/mismatch error, saying that /uibuilder/vendor/vue/dist/vue.min.js is delivered with a "text/html" MIME-Type instead of the correct "text/javascript" MIME Type.

If I am correct, this is due to the configuration of the Express web server, where in order to get a correct MIME type, those static files should be served from a folder configured using the "express.static('public')" function.

However, I could not find out where in Node-Red or in uibuilder the static resources folders (for vue.min.js etc.) are configured.

I have not yet updated Node Red to the latest non-distribution version, however, please let me know if it is of any use here.

Regards,
Ulrich

**Comment by ul-gh** [comment, marked as answer]
OK,

I now uninstalled the Raspberry Pi packaged version of node-red and installed node-red v.3.0.2 via npm.
The VueJS + bootstrap-vue template now runs without an issue so far.

However, if the VueJS + bootstrap-vue template now requires a newer node-red version, it might be worth updating the minimum required version in the release notes in README.md or in the "Using VueJS" section of the same d/nodes/libs/web.js
ocument.

Edit:
Just now, I also found the Express server and Express router setup in file /nodes/libs/web.js, see function serveVendorPackages().
From there, it seems that there is no error.

I guess there is something wrong with the specific node or Express version that comes with the Raspberry Pi OS debian packages, not sure if it is worth looking futher into this..

**Reply by TotallyInformation** [reply]
Hi, sorry for the delayed reply. Just come back from a family holiday travelling across Europe.

uibuilder v5 _should_ still work with the older version of Node-RED. However, I admit that I don't test it. uibuilder v6 WILL however have the minimum node-red version set to v3 and node.js to v14 LTS. For future releases, I'll be trying to align better to the new release schedule of Node-RED and any major upgrades of node.js and node-red will result in the following update of uibuilder being aligned.

> I guess there is something wrong with the specific node or Express version that comes with the Raspberry Pi OS debian packages, not sure if it is worth looking futher into this..

The Pi version will doubtless eventually be updated. However, I would never recommend its use any more that I recommend using any of the other Debian/Rasbian app versions for apps that change rapidly such as node.js, Node-RED, etc. Node-RED in particular is so easy to manage directly either via Dave's install script or installing direct (see my alternate installer repo for example) that it is pretty pointless using the Rasbian repo version.

> Just now, I also found the Express server and Express router setup in file /nodes/libs/web.js, see function serveVendorPackages().
> From there, it seems that there is no error.

That is correct. One of the changes I made a couple of years ago was to tighten the default security when using ExpressJS. That means that uibuilder turns off mime-type sniffing. The consequence of that is that if there is some other error, you may well get the error you were seeing. Though it is more common with CSS files.

With the latest versions of uibuilder, you can of course, make some significant changes to the ExpressJS config, especially if you use a separate web instance which is now allowed. So if you ever get stuck with a setting, that may be a work around.

---

Anyway, I'm now back from hols and a coding break due to an accident that limited my hand movement for a while. So I'm now starting to work on uibuilder v6. This shouldn't have as many breaking changes as v5 (just the node.js and node-red min version updates I hope) but will contain the usual bug fixes and some tweaks such as adding standardised caching options to the experimental uib-list node.

There is a v6 branch if you want to play along :) 

Many thanks for your comments. All comments and ideas are welcome and I will always respond.
